app/src/components/ViewEventsCards/EventCardElement.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useState } from 'react';
+import { IEvent } from 'src/types/event';
+import { Link } from 'react-router-dom';
+import style from './EventCardElement.module.scss';
+import { stringifyDate, isSameDate } from 'src/types/date';
+import { stringifyTime } from 'src/types/time';
+import { viewEventRoute } from 'src/routing';
+import { hasLoaded } from 'src/remote-data';
+import { useNumberOfParticipants, useWaitinglistSpot } from 'src/hooks/cache';
+import { useEditToken, useSavedParticipations } from 'src/hooks/saved-tokens';
+import classNames from 'classnames';
+import { EventState } from 'src/components/ViewEventsCards/EventState';
+import { isInThePast } from 'src/types/date-time';
+import { isNumber } from 'lodash';
+
+interface IProps {
+  eventId: string;
+  event: IEvent;
+}
+
+export const EventCardElement = ({ eventId, event }: IProps) => {
+  const editToken = useEditToken(eventId);
+
+  const { savedParticipations: participationsInLocalStorage } =
+    useSavedParticipations();
+  const participationsForThisEvent = participationsInLocalStorage.filter(
+    (p) => p.eventId === eventId
+  );
+
+  const waitingListSpot = useWaitinglistSpot(
+    eventId,
+    participationsForThisEvent[0]?.email
+  );
+
+  const remoteNumberOfParticipants = useNumberOfParticipants(eventId);
+  const numberOfParticipants = hasLoaded(remoteNumberOfParticipants)
+    ? remoteNumberOfParticipants.data
+    : undefined;
+
+  const numberOfAvailableSpot =
+    event.maxParticipants !== 0 && numberOfParticipants !== undefined
+      ? event.maxParticipants - numberOfParticipants
+      : undefined;
+
+  const registrationState =
+    event.maxParticipants === 0
+      ? 'Plass'
+      : numberOfParticipants === undefined
+      ? 'loading'
```

Comment:
Vil undefined branchen  vises mye? Isåfall så kanskje "Laster" så alt blir norsk 🤔

Reply:
Teksten her vil ikke vises her. Den er på engelsk for å matche svaret som waitingListSpot ka få fra nettverkskallet, slik at laster og loading ikke brukes om hverandre i koden. Hvis state er loading vil "Laster" vises til brukeren
![image](https://user-images.githubusercontent.com/46678893/124431788-206d2980-dd71-11eb-9a84-79cd1a47f46c.png)

Det er også veldig sjeldent loading faktisk sendes videre i lenger tid, den blir borte før man rekker å lese "laster", men jeg får feilmelding om jeg ikke sjekker om variabelen ikke er undefined før jeg leser verdien

Reply:
Eller nvm det var ikke grunnen til at vi brukte loading. Må sjekke med Tarjei hvorfor vi brukte den på engelsk